### PR TITLE
Make sure redux store is passed down the reader

### PR DESCRIPTION
**Pull request by samuelclemens**
In order for react-redux ``connect()`` being able to connect the store to the
desired component it needs to have the redux store on the props/context,
that is being done via the ``<Provider />`` on the root rendered component,
however, doing somewhere in the middle ``ReacDom.render`` breaks the context
passing down the line, I've tried to fix it here specifically for the reader.

Because the the component being rendered is actually the ``<Provder />``, I had to replace direct calling of ``__fullPostInstance.setState()`` to a bit of a callback trickery.

I'm all in for suggestions regarding changing that =)

### CODE REVIEW

File: client/lib/react-helpers/index.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ export default {
 				? document.getElementById( domContainer )
 				: domContainer;
 
+		if ( typeof reduxStore.getState !== 'function'
```
Comment (blowery) [inline client/lib/react-helpers/index.js]: was this just a sanity check? were you having problems with an invalid store being passed?

Comment (samuelclemens) [inline client/lib/react-helpers/index.js]: I've created a separate PR for that: #3028

**Comment by samuelclemens** [comment]
@blowery I've splitted the sanity check to another PR #3028 , anything else I should change here?

**Comment by blowery** [comment]
Sorry for the delay on this. I think I want to hold off a tick and move the full post state into redux instead.

**Comment by nb** [comment]
Isn’t there a way to avoid components like `Dialog` to know `about `context.store`?

**Comment by yurynix** [comment]
@nb Components that doesn't care about ``store`` don't need to know about ``context.store``.
BUT
since ``Dialog`` ``React.render()``s components that do care about ``store`` (anything that has ``connect()`` for example) it must pass it.
Unless we decide to roll our own ``connect()`` that'll take the store not from react's component context but from some other (global?) location.

BTW, that PR might be **breaking** navigation from post view via browser's back/forward buttons. Since @blowery wrote he wouldn't merge that and instead move the full post state into redux, I haven't investigated that any further yet.

**Comment by samuelclemens** [comment]
@nb Components that doesn't care about ``store`` don't need to know about ``context.store``.
BUT
since ``Dialog`` ``React.render()``s components that do care about ``store`` (anything that has ``connect()`` for example) it must pass it.
Unless we decide to roll our own ``connect()`` that'll take the store not from react's component context but from some other (global?) location.

BTW, that PR might be **breaking** navigation from post view via browser's back/forward buttons. Since @blowery wrote he wouldn't merge that and instead move the full post state into redux, I haven't investigated that any further yet.

**Comment by aduth** [comment]
Yeah, this is an unfortunate consequence of us having many `ReactDom.render` scattered throughout the render hierarchy. I encountered this too in #2472, where I came to a near identical solution. Both #2472 and this one had conflicts with #2470.

I can't speak to the integration with Reader, but I've rebased #2472, so we can move forward with the dialog Redux solution there if it's more appropriate.

**Pull request closed without merging**